src/foam/nanos/geocode/GoogleMapsGeocodingDAO.java

```diff
@@ -62,7 +62,7 @@ public FObject put_(X x, FObject obj) {
       if (stored != null && prop_.get(obj) != null ) {
         Address storedAddress = (Address) prop_.get(obj);
         // compare fields that are used to populate Google maps query
-        if ( SafetyUtil.compare(address.getAddress(), storedAddress.getAddress()) == 0 &&
+        if ( SafetyUtil.compare(address.getAddress1(), storedAddress.getAddress1()) == 0 &&
```

Comment:
If the address is structured you should use the streetNumber + " " + streetName instead.